src/storage/src/render/upsert.rs

```diff
@@ -320,18 +252,39 @@ where
     builder.build(move |caps| async move {
         let mut output_cap = caps.into_element();
 
-        let mut snapshot = vec![];
+        let mut state = UpsertState::new(
+            state().await,
+            upsert_shared_metrics,
+            upsert_metrics,
+            source_config.source_statistics,
```

Comment:
dumb q: Since this is inside the operator, should this have been a `source_config.source_statistics.clone()` instead (which returns a `Rc`)? Why or why not?

Reply:
We already cloned it when we we made the `RawSourceConfig` we passed in. Here we are destructing the `source_config` and pulling out the `source_statistics` by value. In rust, you can do this if you only pull each field out at most once, and the struct doesn't implement `Drop`! It's subtle, not a dumb q!!